demystifying_containers/demystifying_containers.md

```diff
@@ -0,0 +1,165 @@
+
+<div class = "care">
+<b style="color: rgb(var(--color-highlight));">A little encouragement...</b><br>
+
+Feeling overwhelmed? Containerization is a topic that can be difficult to wrap your head around at first! Don't worry about the implementation details right now-- this module is just here to give you a little sense of why containers might be useful and what's involved. 
+
+</div>
+
+### Containers vs. virtual machines
+
+![On left side is a schematic of a Docker container with six applications labeled A through F. They sit on top of a Docker layer, which sits on top of a a host operating system layer, which in turn sits on top of an infrastructure layer. On the right side is a schematic of three virtual machines, each with its own guest operating system and application. These three virtual machines sit on top of a hypervisor layer, which sits on top of an infrastructure layer.](media/docker-containerized-and-vm.png)
+
+Some of you might be thinking that containers sound a bit like virtual machines. While virtual machines (or VMs) and containers can serve similar purposes, they aren't exactly the same. 
+
+Virtual machines are divisions of a server into multiple, smaller units. Each machine has its own operating system, and you can divide up hardware elements like CPU and disk space. Virtual machines behave like computers in most respects-- but they're a "virtual" copy.
+
+Containers are much smaller than virtual machines. The element that is "virtualized" is the software, not the hardware underneath (that all comes from your computer). Because of this, it is easier to share and modify container images. 
+
+### Quiz: How containers work
+
+Which of the following describes what a container image is?
+
+[(X)] A set of instructions that describes how a container will be built.
+[( )] A small package of software that has everything it needs to run in any environment.
+[( )] A digital copy of a computer that runs on a server. 
+***
+<div class = "answer">
+
+Unlike a container, which is a lightweight bundle of software with all of the code and dependencies required to run in any environment, a container image is the set of instructions that describes how that container will be built. A digital copy of a computer that runs on a server is a virtual machine.  
+
+</div>
+***
+
+## Container technology
+
+There are a variety of choices when it comes to containerization technology, but the most popular is **Docker**. [Docker's website](https://www.docker.com/) has many resources and articles to help you get started, as well as a [registry for container images, called DockerHub](https://hub.docker.com/), where you can store your own container images and make use of images already out there. 
+
+However there are other providers that you might come across, that are very similar to Docker but might have subtle differences that may make them better for specific workflows (don't worry about those differences right now though). Here are just a few:
+
```

Comment:
Are you able to provide any context about the workflows for which these other tools might be relevant? Like "If you do omics work, you might prefer Buildah, and if you do image analysis you might prefer rkt." Or something.
If not, I think it may be helpful to add a note to give folks more context for how to think about these (e.g. "These container providers all work pretty similarly. In general, you'll want to pick the container provider that seems the most widely used in your field; for most researchers that will be Docker, but if you tend to see another type of container instead, know that the principles outlined in this lesson still apply, even if the implementation might be a little different." Although I definitely just made that up, so don't say it if it's not true! ;) )

Reply:
So it seems (at least in my brief searching around) that most folks either use docker or one of the cloud options, but I'd be uncomfortable making a blanket statement. And I don't think I'd want to try to make a statement about specific workloads (I haven't found a resource that spells it out really, and honesty a lot of the differences are pretty small and get a lot into some computer science weeds that I think might be confusing). Let me see if I can come up with something better than I have now though.

Reply:
Okay, I've added some more stuff here (it's a little hand-wavy, but possible enough?)